test/testcommon/src/main/java/io/pravega/test/common/TestUtils.java

```diff
@@ -41,7 +41,7 @@ public class TestUtils {
 
     // We use a random start position here to avoid ports conflicts when this method is executed from multiple processes
     // in parallel. This is needed since the processes will contend for the same port sequence.
-    private static final AtomicInteger NEXT_PORT = new AtomicInteger(RAND.nextInt());
+    private static final AtomicInteger NEXT_PORT = new AtomicInteger(RAND.nextInt(MAX_PORT_COUNT));
```

Comment:
While perfectly valid, I don't think this change will fix the issue mentioned, as the modulo operation happens on line 53 anyway.

Reply:
RAND.nextInt(MAX_PORT_COUNT) = guarantees that random number generated is always a positive int between 0 and MAX_PORT_COUNT